sympy/ntheory/generate.py

```diff
@@ -421,23 +430,45 @@ def prime(nth):
     n = as_int(nth)
     if n < 1:
         raise ValueError("nth must be a positive integer; prime(1) == 2")
+
+    # Check if n is within the sieve range
     if n <= len(sieve._list):
         return sieve[n]
 
     from sympy.functions.elementary.exponential import log
     from sympy.functions.special.error_functions import li
-    a = 2 # Lower bound for binary search
-    # leave n inside int since int(i*r) != i*int(r) is not a valid property
-    # e.g. int(2*.5) != 2*int(.5)
-    b = int(n*(log(n) + log(log(n)))) # Upper bound for the search.
+    # Use approximations for smaller values of n to avoid li computation
+    if n < 1000:  # Threshold for approximation can be adjusted
+        # Extend sieve up to 8*n as this is empirically sufficient
+        sieve.extend(8 * n)
+        return sieve[n]
 
+    # Dynamically extend the sieve up to an estimated upper bound
```

Comment:
The comment does not accurately reflect the current behavior.

Reply:
I’m sorry, that’s an old unnecessary comment that I didn’t remove. Cleaned it up.